sql/core/src/main/scala/org/apache/spark/sql/execution/SparkOptimizer.scala

```diff
@@ -31,12 +32,14 @@ import org.apache.spark.sql.execution.python.{ExtractGroupingPythonUDFFromAggreg
 class SparkOptimizer(
     catalogManager: CatalogManager,
     catalog: SessionCatalog,
-    experimentalMethods: ExperimentalMethods)
+    experimentalMethods: ExperimentalMethods,
+    spark: SparkSession)
```

Comment:
We can't add `SparkSession`.

Reply:
Any advice on how to get around this?

Reply:
I think you should put `ExecuteUncorrelatedScalarSubquery` into AQE, so we can get session from `AdaptiveExecutionContext`.